util/telemetry/exporter_prometheus.go

```diff
@@ -53,7 +54,8 @@ func (config *Config) port() int {
 
 // RunPrometheusServer starts a prometheus metrics server
 // If 'isDummy' is set to true, the dummy metrics server will be started. If it's false, the prometheus metrics server will be started
-func (m *Metrics) RunPrometheusServer(ctx context.Context, isDummy bool) {
+func (m *Metrics) RunPrometheusServer(ctx context.Context, isDummy bool, wg *sync.WaitGroup) {
+	defer wg.Done()
```

Comment:
instead of adding `Done()` here, could you wrap the call of `RunPrometheusServer` with a `wg.Done()`? that would keep the encapsulation. See below comment for concrete suggestion

Reply:
You're right, that' a bit simpler. Done in 5dbe063d15a3dc6bf66ccd5151a55eff042c0db1